heimdall.google-addon/execution.html

```diff
@@ -16,72 +16,95 @@
             crossorigin="anonymous"></script>
     <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
 </head>
-
 <script>
-    
-    var list_of_ids;
-    
-    function env(value){
-       if(value == 'Test-onDemand'){
-           document.getElementById('addbranch').style="visibility: display; height: 50px;width: 300px;";      
-       }
-       else{
-            document.getElementById('addbranch').style="visibility: hidden; height: 0px;width: 0px;";
-       }
-    }
-    
-    function fun(){
-        var env=$("#environment").val();
-        var filter=$("#selectFilter").val();
-        var suites;
-        var tags;
-        if(filter == "suite"){
-           suites=$("#suitesMultiSelect").val();
-           tags = list_of_ids;
-        }else{
-           tags = "@" + filter;
-        }
+
+   var list_of_ids;
+
+   function env(value) {
+      if (value == 'Test-onDemand') {
+         document.getElementById('addbranch').style = "visibility: display; height: 50px;width: 300px;";
+      }
+      else {
+         document.getElementById('addbranch').style = "visibility: hidden; height: 0px;width: 0px;";
+      }
+   };
+
+
+   function fun() {
+      var env = (document.getElementById("customEnv").value == "") ? $("#environment").val() : document.getElementById("customEnv").value;
+      var suites;
+      var inputTag = document.getElementById('customTag').value;
+      var checkedBoxes = document.getElementsByClassName("custom-control-input");
+      var tags = "";
+
+      for (var i = 0; i < 3; i++) {
+         if (checkedBoxes[i].checked == true) {
+            tags = tags + ",@" + checkedBoxes[i].name;
+         }
+      }
+      tags = tags.substring(1, tags.length);
+
+      if (checkedBoxes[3].checked == true) {
+         suites = $("#suitesMultiSelect").val();
```

Comment:
`suites` is not used below. As I see `$("#suitesMultiSelect").val()` directly being accessed. 
As well  I see multiple ways of  accessing an element,  one via `JQuery $` and another by `document.getElementById`

@vaibsaraswat can we get this discussed across the team to use one norm?

Reply:
Done